### ISSUE TITLE
bug: 

### ISSUE THREAD

**Issue by codifierr** [apache/apisix#9941]
### Current Behavior

The request-id plugin doesn't work properly when the range-id algorithm is given to generate the request-id.

Apisix responds with 500 internal server errors.

### Expected Behavior

Request-id plugin should work with range_id algorithm as it is supported by apisix 3.3.
https://apisix.apache.org/docs/apisix/3.3/plugins/request-id/

### Error Logs

apisix-85f7854f87-jmkqg apisix 2023/08/01 07:29:41 [error] 49#49: *635967 lua entry thread aborted: runtime error: /usr/local/apisix/apisix/plugins/request-id.lua:230: attempt to index local 'range_id' (a nil value)
apisix-85f7854f87-jmkqg apisix stack traceback:
apisix-85f7854f87-jmkqg apisix coroutine 0:
apisix-85f7854f87-jmkqg apisix 	/usr/local/apisix/apisix/plugins/request-id.lua: in function 'get_request_id'
apisix-85f7854f87-jmkqg apisix 	/usr/local/apisix/apisix/plugins/request-id.lua:257: in function 'phase_func'
apisix-85f7854f87-jmkqg apisix 	/usr/local/apisix/apisix/plugin.lua:1097: in function 'run_plugin'
apisix-85f7854f87-jmkqg apisix 	/usr/local/apisix/apisix/init.lua:633: in function 'http_access_phase'

### Steps to Reproduce

1.) Run apisix
2.) create a route with request-id plugin and provide the algorithm as range_id.

### Environment

- APISIX version (run `apisix version`): 3.3.0
- Operating system (run `uname -a`): Linux apisix-85f7854f87-jmkqg 6.1.30-0-virt #1-Alpine SMP Fri, 26 May 2023 06:53:59 +0000 aarch64 GNU/Linux
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`):
nginx version: openresty/1.21.4.1
built by gcc 10.2.1 20210110 (Debian 10.2.1-6)
built with OpenSSL 1.1.1s  1 Nov 2022
TLS SNI support enabled
configure arguments: --prefix=/usr/local/openresty/nginx --with-cc-opt='-O2 -DAPISIX_BASE_VER=1.21.4.1.8 -DNGX_GRPC_CLI_ENGINE_PATH=/usr/local/openresty/libgrpc_engine.so -DNGX_HTTP_GRPC_CLI_ENGINE_PATH=/usr/local/openresty/libgrpc_engine.so -DNGX_LUA_ABORT_AT_PANIC -I/usr/local/openresty/zlib/include -I/usr/local/openresty/pcre/include -I/usr/local/openresty/openssl111/include' --add-module=../ngx_devel_kit-0.3.1 --add-module=../echo-nginx-module-0.62 --add-module=../xss-nginx-module-0.06 --add-module=../ngx_coolkit-0.2 --add-module=../set-misc-nginx-module-0.33 --add-module=../form-input-nginx-module-0.12 --add-module=../encrypted-session-nginx-module-0.09 --add-module=../srcache-nginx-module-0.32 --add-module=../ngx_lua-0.10.21 --add-module=../ngx_lua_upstream-0.07 --add-module=../headers-more-nginx-module-0.33 --add-module=../array-var-nginx-module-0.05 --add-module=../memc-nginx-module-0.19 --add-module=../redis2-nginx-module-0.15 --add-module=../redis-nginx-module-0.3.9 --add-module=../ngx_stream_lua-0.0.11 --with-ld-opt='-Wl,-rpath,/usr/local/openresty/luajit/lib -Wl,-rpath,/usr/local/openresty/wasmtime-c-api/lib -L/usr/local/openresty/zlib/lib -L/usr/local/openresty/pcre/lib -L/usr/local/openresty/openssl111/lib -Wl,-rpath,/usr/local/openresty/zlib/lib:/usr/local/openresty/pcre/lib:/usr/local/openresty/openssl111/lib' --add-module=/tmp/tmp.aLb1NUnBtM/openresty-1.21.4.1/../mod_dubbo-1.0.2 --add-module=/tmp/tmp.aLb1NUnBtM/openresty-1.21.4.1/../ngx_multi_upstream_module-1.1.1 --add-module=/tmp/tmp.aLb1NUnBtM/openresty-1.21.4.1/../apisix-nginx-module-1.12.0 --add-module=/tmp/tmp.aLb1NUnBtM/openresty-1.21.4.1/../apisix-nginx-module-1.12.0/src/stream --add-module=/tmp/tmp.aLb1NUnBtM/openresty-1.21.4.1/../apisix-nginx-module-1.12.0/src/meta --add-module=/tmp/tmp.aLb1NUnBtM/openresty-1.21.4.1/../wasm-nginx-module-0.6.4 --add-module=/tmp/tmp.aLb1NUnBtM/openresty-1.21.4.1/../lua-var-nginx-module-v0.5.3 --add-module=/tmp/tmp.aLb1NUnBtM/openresty-1.21.4.1/../grpc-client-nginx-module-v0.4.2 --with-poll_module --with-pcre-jit --with-stream --with-stream_ssl_module --with-stream_ssl_preread_module --with-http_v2_module --without-mail_pop3_module --without-mail_imap_module --without-mail_smtp_module --with-http_stub_status_module --with-http_realip_module --with-http_addition_module --with-http_auth_request_module --with-http_secure_link_module --with-http_random_index_module --with-http_gzip_static_module --with-http_sub_module --with-http_dav_module --with-http_flv_module --with-http_mp4_module --with-http_gunzip_module --with-threads --with-compat --with-stream --with-http_ssl_module
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`):
{"hostname":"apisix-85f7854f87-jmkqg","version":"3.3.0","id":"2a6ca3a3-5813-4a29-a850-34d95cebf6c5","boot_time":1690869860,"etcd_version":"3.5.0"}
- APISIX Dashboard version, if relevant: NA
- Plugin runner version, for issues related to plugin runners:
- LuaRocks version, for installation issues (run `luarocks --version`): NA

**Comment by shreemaan-abhishek** [comment]
Please share your route configuration.

**Comment by codifierr** [comment]
`apiVersion: apisix.apache.org/v2
kind: ApisixRoute
metadata:
  name: http-ping-route
spec:
  http:
    - name: http-ping-rule
      match:
        paths:
        - /ping
      backends:
        - serviceName: a-http-ping
          servicePort: 8080
      plugins:
      - name: limit-count
        enable: true
        config:
          count: 20
          time_window: 1
          key: remote_addr
      - name: request-id
        enable: true
        config:
          algorithm: range_id
          length: 16
      - name: key-auth
        enable: true`

**Comment by codifierr** [comment]
@shreemaan-abhishek  Yes this worked. But since it is just empty IMO we should add this as default or document it somewhere.

**Comment by shreemaan-abhishek** [comment]
Yes, I just raised a PR.

**Comment by davidrjonas** [comment]
This appears to still be an issue in 3.9.1.

```
$ docker compose config | rg apache/apisix
    image: apache/apisix:3.9.1-debian

$ rg -A 2 request-id ./config/apisix.yaml 
11:      # https://apisix.apache.org/docs/apisix/plugins/request-id/
12:      request-id:
13-        algorithm: range_id
14-

$ curl -ksSf https://api.test/
curl: (22) The requested URL returned error: 500

$ docker compose logs api | rg -A 9 bad
sandbox-api-1  | 2024-06-21T23:34:42.613238033Z 2024/06/21 23:34:42 [error] 49#49: *683 lua entry thread aborted: runtime error: /usr/local/apisix/apisix/plugins/request-id.lua:71: bad argument #2 to 'new' (cannot convert 'nil' to 'int')
sandbox-api-1  | 2024-06-21T23:34:42.613289498Z stack traceback:
sandbox-api-1  | 2024-06-21T23:34:42.613297875Z coroutine 0:
sandbox-api-1  | 2024-06-21T23:34:42.613304514Z 	[C]: in function 'new'
sandbox-api-1  | 2024-06-21T23:34:42.613311747Z 	/usr/local/apisix/apisix/plugins/request-id.lua:71: in function 'get_request_id'
sandbox-api-1  | 2024-06-21T23:34:42.613318579Z 	/usr/local/apisix/apisix/plugins/request-id.lua:98: in function 'phase_func'
sandbox-api-1  | 2024-06-21T23:34:42.613325345Z 	/usr/local/apisix/apisix/plugin.lua:1155: in function 'run_plugin'
sandbox-api-1  | 2024-06-21T23:34:42.613332133Z 	/usr/local/apisix/apisix/plugin.lua:1222: in function 'run_global_rules'
sandbox-api-1  | 2024-06-21T23:34:42.613338711Z 	/usr/local/apisix/apisix/init.lua:686: in function 'http_access_phase'
sandbox-api-1  | 2024-06-21T23:34:42.613345389Z 	access_by_lua(nginx.conf:297):2: in main chunk, client: 172.21.0.16, server: _, request: "GET / HTTP/1.0", host: "api.test"
```

Adding `range_id: {}` to the config fixes the issue,

```
$ rg -A 2 request-id ./config/apisix.yaml
11:      # https://apisix.apache.org/docs/apisix/plugins/request-id/
12:      request-id:
13-        algorithm: range_id
14-        range_id: {}

$ docker compose up --force-recreate api

$ curl -ksSi https://api.test/ | head -n 1
HTTP/1.1 200 OK
```
